doc/languages-frameworks/agda.section.md

```diff
@@ -65,12 +124,67 @@ A derivation can then be written using `agdaPackages.mkDerivation`. This has sim
 * `libraryName` should be the name that appears in the `*.agda-lib` file, defaulting to `pname`.
 * `libraryFile` should be the file name of the `*.agda-lib` file, defaulting to `${libraryName}.agda-lib`.
 
+Here is an example `default.nix`
+
+```
+{ nixpkgs ?  <nixpkgs> }:
+with (import nixpkgs {});
+agdaPackages.mkDerivation {
+  version = "1.0";
+  pname = "my-agda-lib";
+  src = ./.;
+  buildInputs = [
+    agdaPackages.standard-library
+  ];
+}
+```
+
 ### Building Agda packages
 The default build phase for `agdaPackages.mkDerivation` simply runs `agda` on the `Everything.agda` file.
 If something else is needed to build the package (e.g. `make`) then the `buildPhase` should be overridden.
 Additionally, a `preBuild` or `configurePhase` can be used if there are steps that need to be done prior to checking the `Everything.agda` file.
 `agda` and the Agda libraries contained in `buildInputs` are made available during the build phase.
 
+For example to generate an `Everything.agda` before checking you could add a `preBuild` phase
```

Comment:
I agree that this is quite an extensive example. I would recommend:

* Make this example a library function and use it in at least one actual library that doesn't have an `Everything.agda` file. Possibly in a separate MR.
* Reduce the example to the case of assuming and using a simple makefile. One usecase of using `preBuild` is typically when someone has a library with a makefile or some other existing mechanism to create an `Everything.agda` (like the standard library does) and they just want to drop that mechanism into here.
* Alternative: Instead of a makefile you could maybe just say in words that `preBuild` can be used to create the everything file, and then link to some agdaPackages code examples where this is done.

Reply:
I would go for a separate MR.